modules/business-text/business-text.service.js

```diff
@@ -37,32 +39,122 @@ class BusinessTextService {
       };
     }
 
-    const text = await BusinessText.findByIdAndUpdate(id, businessText, {
+    const newPage = files.length
+      ? await this.pageImageReplaceHandler(businessText, files)
+      : businessText;
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const newImages = this.imageFinder(newPage);
+      const oldImages = this.imageFinder(oldPage);
+
+      const imagesToDelete = oldImages.filter(
+        img => newImages.findIndex(newImg => newImg === img) === -1
+      );
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
+      }
+    } else {
+      throw new Error(BUSINESS_TEXT_NOT_FOUND);
+    }
+
+    const text = await BusinessText.findByIdAndUpdate(id, newPage, {
       new: true,
     });
 
     return text || null;
   }
 
-  async addBusinessText(data) {
-    const pages = await this.checkBusinessTextExistByCode(data);
+  async addBusinessText(businessText, files) {
+    const existingPages = await this.checkBusinessTextExistByCode(businessText);
 
-    if (pages.length) {
+    if (existingPages.length) {
       throw new Error(BUSINESS_TEXT_ALREADY_EXIST);
     }
-    return new BusinessText(data).save();
+
+    let newPage = '';
+    if (files.length) {
+      newPage = await this.pageImageReplaceHandler(businessText, files);
+    }
+
+    return new BusinessText(newPage || businessText).save();
   }
 
   async deleteBusinessText(id) {
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const imagesToDelete = this.imageFinder(oldPage);
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
```

Comment:
If we failed to delete the images then we should return the error to the client and ask him to try it again. I don't think that frontend somehow handles the case when we failed to delete images but delete other data from the database successfully

Reply:
already added error when some of images are fails with deleting